libnavigation-base/src/main/java/com/mapbox/navigation/base/utils/DecodeUtils.kt

```diff
@@ -0,0 +1,55 @@
+@file:JvmName("DecodeUtils")
+
+package com.mapbox.navigation.base.utils
+
+import com.mapbox.api.directions.v5.DirectionsCriteria
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.core.constants.Constants
+import com.mapbox.geojson.LineString
+import com.mapbox.geojson.Point
+import com.mapbox.geojson.utils.PolylineUtils
+import com.mapbox.navigation.base.internal.utils.CacheResultUtils.cacheResult
+
+private val polylineDecodeCache = PolylineUtils::decode.cacheResult(maxSize = 50)
+
+fun DirectionsRoute.completeGeometryToLinestring(): LineString? {
+    return geometry()?.geometryToLinestring(precision())
+}
+
+fun DirectionsRoute.completeGeometryToPoints(): List<Point>? {
+    return geometry()?.geometryToPoints(precision())
+}
+
+fun DirectionsRoute.stepsGeometryToLinestring(): List<List<LineString?>?>? {
+    val precision = precision()
+    return legs()?.map { leg ->
+        leg.steps()?.map { step ->
+            step.geometry()?.geometryToLinestring(precision)
+        }
+    }
+}
+
+fun DirectionsRoute.stepsGeometryToPoints(): List<List<List<Point>?>?>? {
+    val precision = precision()
+    return legs()?.map { leg ->
+        leg.steps()?.map { step ->
+            step.geometry()?.geometryToPoints(precision)
+        }
+    }
+}
+
+fun String.geometryToLinestring(precision: Int): LineString {
```

Comment:
I don't think we should create an extension function on strings. This use case is specific to Mapbox and doesn't apply to most strings.

Reply:
The functions should also probably be wrapped in and `object SomeName {}` so there is a namespace.

Reply:
We have a public `ReplayRouteFactory.mapGeometry` function. It accepts an arbitrary `String`, that could be a geometry of a route or a step. I guess we won't use caching in this case, right?

Reply:
@LukasPaczos @cafesilencio

Reply:
I don't think we need to cover this scenario. Replayer is a debug tool anyway.